### DISCUSSION TITLE
Interceptors seems to trigger on runSelect only ?

### DISCUSSION THREAD

**Discussion by Ninja4Panda** [simolus3/drift#2850] [Q&A] [answered]
Hey guys new here so thanks in advance! 
I was trying to save logs to the database before every inserts statement so I added my own interceptor like so just to test it out.
![image](https://github.com/simolus3/drift/assets/61926182/983d53a2-d891-412c-9974-710dc9ede8c9)

This is my database init, it is pretty much the same as the snippet from the tutorial but I have added a logging interceptor on it 
![image](https://github.com/simolus3/drift/assets/61926182/21ac7e25-f6bc-4977-b04f-029025eb90fd)
 
However, I realised that the interceptor is triggering with the runSelect instead of runInsert. Did I misunderstand something ? 
![image](https://github.com/simolus3/drift/assets/61926182/f4b7ce91-19d1-42a1-83f8-3d443dcaba1e)

**Comment by simolus3** [comment, marked as answer]
The way you have it configured, the interceptor should also be applied for inserts (and all other kinds of statements).
The reason `runSelect` is called for inserts is because the insert is using `RETURNING`. That requires the database to return rows to drift, which is only possible with `runSelect`. Updates and deletes with a returning clause would also use `runSelect`. You should see calls to `runInsert` for inserts without a returning clause.

**Reply by Ninja4Panda** [reply]
Thanks for the reply! Amazing library btw! 
Understood, it is working without the RETURNING now. 
Is there a reason behind this ? Cause technically insert happened right ?

**Reply by simolus3** [reply]
> Cause technically insert happened right ?

Right, but that doesn't matter since sqlite3 uses a uniform API that works for all kinds of statements. The different methods just indicate what kind of result they should return.
I think the names used by drift here are just somewhat poorly chosen. They made sense originally, but when `RETURNING` was introduced it stirred things up a bit. `runSelect` should be called something like `runWithRows`. `runInsert` should be `runAndReturnLastInsertRowid` and `runUpdate` and `runDelete` should both be `runAndReturnAffectedRows`.
